rest/route/host_agent_test.go

```diff
@@ -857,295 +858,656 @@ func TestTaskLifecycleEndpoints(t *testing.T) {
+			d.Tq1.Queue = append(d.Tq1.Queue, model.TaskQueueItem{Id: "tg1-task3"})
+			d.Tq1.DistroQueueInfo = model.DistroQueueInfo{
+				Length:         3,
+				TaskGroupInfos: []model.TaskGroupInfo{{Name: "task-group-1", Count: 3}},
+			}
+			require.NoError(t, d.Tq1.Save())
+			tg1Task3 := &task.Task{
+				Id:                "tg1-task3",
+				Status:            evergreen.TaskUndispatched,
+				Activated:         true,
+				Project:           d.Project1.Id,
+				StartTime:         utility.ZeroTime,
+				TaskGroup:         d.Tg1Task1.TaskGroup,
+				Version:           d.Version1.Id,
+				BuildId:           d.BuildVariant1.Id,
+				BuildVariant:      d.BuildVariant1.BuildVariant,
+				TaskGroupMaxHosts: 2,
+			}
+			require.NoError(t, tg1Task3.Insert())
+			require.NoError(t, task.UpdateOne(bson.M{"_id": d.Tg1Task1.Id},
+				bson.M{"$set": bson.M{"task_group_max_hosts": 2}}))
+			require.NoError(t, task.UpdateOne(bson.M{"_id": d.Tg1Task2.Id},
+				bson.M{"$set": bson.M{"task_group_max_hosts": 2}}))
+			details := &apimodels.GetNextTaskDetails{}
+			// The first host should get the top of the task group.
+			nextTaskId := d.Tq1.Queue[0].Id
+			task, shouldTeardown, err := assignNextAvailableTask(ctx, env, d.Tq1, model.NewTaskDispatchService(time.Minute), d.Host1, details)
+			require.NoError(t, err)
+			require.NotNil(t, task)
+			assert.False(t, shouldTeardown)
+			assert.Equal(t, nextTaskId, task.Id)
+
+			tq, err := model.LoadTaskQueue(d.Distro1.Id)
+			require.NoError(t, err)
+			assert.Equal(t, 2, tq.Length())
+
+			h, err := host.FindOneId(ctx, d.Host1.Id)
+			require.NoError(t, err)
+			assert.Equal(t, nextTaskId, h.RunningTask)
+
+			// The second host should get the next task of the task group.
+			nextTaskId = d.Tq1.Queue[0].Id
+			task, shouldTeardown, err = assignNextAvailableTask(ctx, env, d.Tq1, model.NewTaskDispatchService(time.Minute), d.Host2, details)
+			require.NoError(t, err)
+			require.NotNil(t, task)
+			assert.False(t, shouldTeardown)
+			assert.Equal(t, nextTaskId, task.Id)
+
+			tq, err = model.LoadTaskQueue(d.Distro1.Id)
+			require.NoError(t, err)
+			assert.Equal(t, 1, tq.Length())
+
+			h, err = host.FindOneId(ctx, d.Host2.Id)
+			require.NoError(t, err)
+			assert.Equal(t, nextTaskId, h.RunningTask)
+
+			// The third host should not get a task, since the limit is 2.
+			task, shouldTeardown, err = assignNextAvailableTask(ctx, env, d.Tq1, model.NewTaskDispatchService(time.Minute), extraHost, details)
+			fmt.Print(task, shouldTeardown, err)
+			require.NoError(t, err)
+			assert.Nil(t, task)
+			assert.False(t, shouldTeardown)
+
+			tq, err = model.LoadTaskQueue(d.Distro1.Id)
+			require.NoError(t, err)
+			assert.Equal(t, 0, tq.Length())
+
+			h, err = host.FindOneId(ctx, extraHost.Id)
+			require.NoError(t, err)
+			assert.Equal(t, "", h.RunningTask)
+		},
+	} {
+		t.Run(fmt.Sprintf("%s with %s dispatcher settings", tName, evergreen.DispatcherVersionRevisedWithDependencies), func(t *testing.T) {
```

Comment:
it feels like we can just print the task name, if we're testing with this dispatcher setting every time, particularly since you mentioned above that we only have one setting now.